src/flow/assembly_mh.hh

```diff
@@ -358,11 +358,12 @@ protected:
         double conduct =  ad_->conductivity.value(ele.centre(), ele);
 	    double bet =  ad_->beta.value(ele.centre(), ele);
 	    auto w = ad_->field_ele_velocity.value(ele.centre(), ele);
-        double scale = 1 / cs /conduct + bet * arma::norm(w) / cs;
+        auto an = ad_->anisotropy.value(ele.centre(), ele );
+        arma::mat33 scale = an.i() / cs / conduct + bet * arma::norm(w) * arma::eye(3,3) / cs; //inverze an
         assemble_sides_scale(dh_cell, scale);
     }
     
-    void assemble_sides_scale(const DHCellAccessor& dh_cell, double scale)
+    void assemble_sides_scale(const DHCellAccessor& dh_cell, arma::mat33 scale)
```

Comment:
Rather declare "scale" as reference to const (const arma::mat33& scale). It avoids additional allocation of temporary matrix object. Applies also to the same method in class AssemblyMH_Newton.